app/code/local/Magmodules/Webwinkelconnect/Model/Observer.php

```diff
@@ -164,25 +164,62 @@ public function sendSyncUrl(): void {
             return;
         }
 
-        $curl = new Varien_Http_Adapter_Curl();
-        $curl->setConfig([
-            'timeout'   => 10,
-        ]);
         $url = 'https://dashboard.webwinkelkeur.nl/webshops/sync_url';
         $data = json_encode([
             'webshop_id' => $helper->getShopId(),
             'api_key' => $helper->getApiKey(),
             'url' => Mage::getBaseUrl(Mage_Core_Model_Store::URL_TYPE_WEB) . 'webwinkelkeur/index/sync',
         ]);
+
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_FAILONERROR, true);
+        curl_setopt( $ch, CURLOPT_HTTPHEADER, array('Content-Type:application/json'));
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_POST, true);
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
         try {
-            $curl->addOption(CURLOPT_FAILONERROR, 1);
-            $curl->write(Zend_Http_Client::POST, $url, '2', ['Content-Type:application/json'], $data);
-            $response = $curl->read();
-            if (!$response) {
-                throw new Exception(sprintf('Could not send sync URL to dashboard: (%s) %s',$curl->getErrno(), $curl->getError()));
+            $response = curl_exec($ch);
+            if ($response === false) {
+                throw new Exception(sprintf('Could not send sync URL to dashboard: %s', curl_error($ch)));
             }
         } finally {
-            $curl->close();
+            curl_close($ch);
         }
     }
+
+
+    public function addOrderDataJsonThankYouPage(Varien_Event_Observer $observer): void {
+        if (Mage::app()->getFrontController()->getAction()->getFullActionName() != 'checkout_onepage_success') {
+            return;
+        }
+
+        if (!Mage::getStoreConfig('webwinkelconnect/privacy_first_option/privacy_popup')) {
+            return;
+        }
+
+        $order_id = $observer->getEvent()->getOrderIds()[0];
+        if (!$order_id) {
+            return;
+        }
+
+        $order = Mage::getModel('sales/order')->load($order_id);
+        $order_data = [
+            'webshopId' => Mage::helper('webwinkelconnect')->getShopId(),
+            'orderNumber' => $order->getIncrementId(),
+            'email' => $order->getCustomerEmail(),
+            'firstName' => $order->getData('customer_firstname'),
+            'inviteDelay' => Mage::getStoreConfig('webwinkelconnect/invitation/delay'),
+        ];
+        try {
+            $order_data['signature'] = Mage::helper('webwinkelconnect/Hash')->getHashForDash($order_data);
+        } catch (Magmodules_Webwinkelconnect_Exception $e) {
+            Mage::logException($e);
```

Comment:
Does it make sense to continue in this case? Order data without signature won't work, or does it?